.github/workflows/sync-gitster-git.yml

```diff
@@ -0,0 +1,120 @@
+name: sync-gitster-git-branches
+
+on:
+  schedule:
+    - cron: '17 6 * * *'
+  workflow_dispatch:
+
+env:
+  SOURCE_REPOSITORY: gitster/git
+  TARGET_REPOSITORY: gitgitgadget/git
+
+jobs:
+  sync-gitster-git-branches:
+    runs-on: ubuntu-latest
+    steps:
+      - name: check which refs need to be synchronized
+        uses: actions/github-script@v6
+        id: check
+        with:
+          script: |
+            const sleep = async (milliseconds) => {
+              return new Promise(resolve => setTimeout(resolve, milliseconds))
+            }
+
+            const getRefs = async (repository) => {
+              for (let i = 1; i <= 10; i++) {
+                try {
+                  const [owner, repo] = repository.split('/')
+                  const { data } = await github.rest.git.listMatchingRefs({
+                    owner,
+                    repo,
+                    ref: 'heads/'
```

Comment:
How about `'heads/maint-'` to reduce the volume of refs returned?

Reply:
I would _love_ to reduce the volume, but `heads/maint-*` is too restrictive: `/^refs\/heads\/(maint-\d|[a-z][a-z]\/)/` matches not only `maint-*` but also `[a-z][a-z]/*`, i.e. branches like [`ts/unpacklimit-config-fix`](https://github.com/gitster/git/tree/ts/unpacklimit-config-fix). And that's on purpose: contributors frequently base their contributions directly on top of other patch series that are in flight, where having them as PR target branches in `gitgitgadget/git` really helps.

Reply:
Sorry, I missed the `|` in the regex.

Reply:
> Sorry, I missed the `|` in the regex.

In hindsight, I see how easily it could be missed. I hope the code comment will help in the future.